### PR TITLE
REST APIs for Rating and Commenting an API

### PR DESCRIPTION
**Pull request by ChamNDeSilva**
This PR may contain methods which are not yest fully implemented. 
Please note that this is a feature which is not fully completed at the moment when this PR is sent.

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/impl/APIStoreImpl.java [hunk 1]
```diff
@@ -425,6 +430,29 @@ public Policy getPolicy(String policyLevel, String policyName) throws APIManagem
     }
 
     @Override
+    public Comment getCommentByUUID(String commentId, String apiId) throws APIManagementException {
+        Comment comment;
+        try {
+            comment = getCommentDAO().getCommentByUUID(commentId, apiId);
+        } catch (APIMgtDAOException e) {
+            String errorMsg = "Error occurred while retrieving label information";
```
Comment (prasa7) [inline components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/impl/APIStoreImpl.java]: Add context information for the log message. You can add "apiId" and "commentId" to the log message.

File: components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/dao/impl/CommentDAO.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
+/*
+ *
+ *   Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (ChamNDeSilva) [inline components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/dao/impl/CommentDAO.java]: Fixed.

File: components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/dao/impl/CommentDAOImpl.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+import org.wso2.carbon.apimgt.core.models.Comment;
+
+/**
+ * Default implementation of the CommentDAO interface. Uses SQL syntax that is common to H2 and MySQL DBs.
+ * Hence is considered as the default due to its re-usability.
+ */
+public class CommentDAOImpl implements CommentDAO {
+
+
```
Comment (ChamNDeSilva) [inline components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/dao/impl/CommentDAOImpl.java]: Fixed.

File: features/apimgt/org.wso2.carbon.apimgt.core.feature/resources/dbscripts/h2.sql [hunk 4]
```diff
@@ -392,3 +392,28 @@ CREATE TABLE IF NOT EXISTS AM_WORKFLOWS (
...
+  `CREATED_BY` VARCHAR(100),
+  `CREATED_TIME` TIMESTAMP(6) DEFAULT CURRENT_TIMESTAMP(6),
+  `UPDATED_BY` VARCHAR(100),
+  `LAST_UPDATED_TIME` TIMESTAMP(6) DEFAULT CURRENT_TIMESTAMP(6),
+  FOREIGN KEY(`API_ID`) REFERENCES `AM_API`(`API_ID`) ON UPDATE CASCADE ON DELETE RESTRICT
+);
+
+CREATE TABLE IF NOT EXISTS `AM_API_COMMENTS` (
+  `COMMENT_ID` INTEGER AUTO_INCREMENT,
+  `COMMENT_TEXT` VARCHAR(512),
```
Comment (ChamNDeSilva) [inline features/apimgt/org.wso2.carbon.apimgt.core.feature/resources/dbscripts/h2.sql]: Fixed in all DB schemas.

File: components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/impl/APIStoreImpl.java [hunk 5]
```diff
@@ -96,9 +101,9 @@
```
Comment (ChamNDeSilva) [inline components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/impl/APIStoreImpl.java]: Added method Comments to constructor.

File: components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/impl/APIStoreImpl.java (hunk 1 above)
Comment (ChamNDeSilva) [inline components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/impl/APIStoreImpl.java]: Fixed.

File: components/apimgt/org.wso2.carbon.apimgt.core/src/test/java/org/wso2/carbon/apimgt/core/impl/APIMgtAdminServiceImplTestCase.java [hunk 6]
```diff
@@ -1,110 +0,0 @@
-/*
```
Comment (harsha89) [inline components/apimgt/org.wso2.carbon.apimgt.core/src/test/java/org/wso2/carbon/apimgt/core/impl/APIMgtAdminServiceImplTestCase.java]: Any reason why this call is removed?

Comment (ChamNDeSilva) [inline components/apimgt/org.wso2.carbon.apimgt.core/src/test/java/org/wso2/carbon/apimgt/core/impl/APIMgtAdminServiceImplTestCase.java]: Fixed by re-adding the test case.

File: components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/dao/impl/CommentDAO.java [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+
+package org.wso2.carbon.apimgt.core.dao.impl;
+
+import org.wso2.carbon.apimgt.core.exception.APIMgtDAOException;
+import org.wso2.carbon.apimgt.core.models.Comment;
+
+/**
+ * Provides access to the Comment data layer
+ */
+public interface CommentDAO {
```
Comment (uvindra) [inline components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/dao/impl/CommentDAO.java]: Do we need to add a separate DAO for comments? Comments are against a given API, so you can include this in the existing ApiDAO itself, no need to add a separate DAO for this

File: components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/dao/impl/RatingDAO.java [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+ *  specific language governing permissions and limitations
+ *  under the License.
+ *
+ */
+package org.wso2.carbon.apimgt.core.dao.impl;
+
+/**
+ * Provides access to the Rating data layer
+ */
+public interface RatingDAO {
```
Comment (uvindra) [inline components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/dao/impl/RatingDAO.java]: Do we need to add a separate DAO for ratings? Ratings are against a given API, so you can include this in the existing ApiDAO itself, no need to add a separate DAO for this

File: components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/dao/impl/CommentDAO.java (hunk 7 above)
Comment (ChamNDeSilva) [inline components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/dao/impl/CommentDAO.java]: Removed CommentDAO class and used ApiDAO class for DAO operations of Comments.

File: components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/dao/impl/RatingDAO.java (hunk 8 above)
Comment (ChamNDeSilva) [inline components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/dao/impl/RatingDAO.java]: Removed RatingDAO class and used ApiDAO class for DAO operations of API Ratings

**Pull request merged**